ButtplugServerGUI/WebsocketServerControl.xaml.cs

```diff
@@ -23,17 +23,66 @@ namespace ButtplugServerGUI
     public partial class WebsocketServerControl : UserControl
     {
         private ButtplugWebsocketServer _ws;
+        private ButtplugService _service;
+        private uint _port;
+        private bool _secure;
 
-        public WebsocketServerControl()
+        public WebsocketServerControl(ButtplugService aService)
         {
             InitializeComponent();
             _ws = new ButtplugWebsocketServer();
-            
+            _service = aService;
+            _port = 12345;
+            _secure = false;
         }
 
-        public void StartServer(ButtplugService aService)
+        public void StartServer()
         {
-            _ws.StartServer(aService);
+            _ws.StartServer(_service, (int) _port, _secure);
+            ((Button)ConnToggleButton).Content = "Stop";
+            ((CheckBox)SecureCheckBox).IsEnabled = false;
+            ((TextBox)PortTextBox).IsEnabled = false;
+        }
+        public void StopServer()
+        {
+            _ws.StopServer();
+            ((Button)ConnToggleButton).Content = "Start";
+            ((CheckBox)SecureCheckBox).IsEnabled = true;
+            ((TextBox)PortTextBox).IsEnabled = true;
+        }
+
+        private void ConnToggleButton_Click(object sender, RoutedEventArgs e)
+        {
+
+            if (((Button)ConnToggleButton).Content == "Start")
+            {
+                StartServer();
+            }
+            else
+            {
+                StopServer();
+            }
+        }
+       
+
+        private void PortTextBox_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            if (UInt32.TryParse(((TextBox)PortTextBox).Text, out uint port) && port >= 1024 && port <= 65535)
+            {
+                _port = port;
+                return;
+            }
+            ((TextBox)PortTextBox).Text = _port.ToString();

```

Comment:
Not sure if we want an error text or something here, but I would figure anyone changing this would hopefully at least somewhat know what they're doing. We'll probably find that out when it goes out in the wild heh.